src/checkpoint/container/view_serialize.h

```diff
@@ -324,25 +348,33 @@ inline void serialize_impl(SerializerT& s, Kokkos::DynRankView<T,Args...>& view)
   s | init;
 
   if (init) {
+    auto host_view = Kokkos::create_mirror_view(view);
```

Comment:
We should pass `WithoutInitializing` and maybe an exec space for this

Reply:
In general, `WithoutInitializing` avoids kernel invocation. Allocation inside Kokkos still potentially synchronizes stuff, but easing that is much further out on the horizon.